### PR TITLE
update readme

### PR DESCRIPTION
**Pull request by vasiliy-t**
- update library description
- make references to base collectors
- add brief base collectors description

### CODE REVIEW

**Review by ZolotarevIgor-mail-ru — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -34,15 +43,17 @@ export plugins:
 or you can write your [custom plugin](https://www.tarantool.io/en/doc/latest/book/monitoring/plugins/#writing-custom-plugins) and use it.
 Hopefully, plugins for other TSDBs will be supported soon.
 
-## Examples
+## Metric types
+
+Four types of metrics are offered: Counter, Gauge, Summary and Histogram.
```
Comment (opomuc) [inline README.md:48]: maybe `There are four basic metric collectors available:`?

File: README.md [hunk 2]
```diff
@@ -34,15 +43,17 @@ export plugins:
...
+Four types of metrics are offered: Counter, Gauge, Summary and Histogram. 
+The exact semantics of each metric follows the [prometheus metric types](https://prometheus.io/docs/concepts/metric_types/).
 
-Below are examples of using metrics primitives.
+### Counter
 
-Note that this usage is independent of export-plugins such as
-Prometheus / Graphite / etc.
-For documentation on plugins usage, see [this](https://www.tarantool.io/en/doc/latest/book/monitoring/plugins).
+Counter is a cummulative metric whose value could only be incremented or reset to zero on restart.
```
Comment (opomuc) [inline README.md:53]: `which value can only be...`

File: README.md [hunk 3]
```diff
@@ -53,7 +64,12 @@ local http_requests_total_counter = metrics.counter('http_requests_total')
 http_requests_total_counter:inc(1, {method = 'GET'})
 ```
 
-Using gauges:
+### Gauge
+
+Gauge is a metric that represents a single numerical value that could change arbitrarily.
```
Comment (opomuc) [inline README.md:69]: `could change` -> `may be changed` / `can be changed`

File: README.md [hunk 4]
```diff
@@ -53,7 +64,12 @@ local http_requests_total_counter = metrics.counter('http_requests_total')
...
+Gauges are useful for capturing the snapshot of current state, e.g. CPU utilization, number of open connections.
```
Comment (opomuc) [inline README.md:70]: `a snapshot of the current state`

File: README.md [hunk 5]
```diff
@@ -68,7 +84,15 @@ metrics.register_callback(function()
 end)
 ```
 
-Using histograms:
+### Histogram
+
+Histogram samples observations and counts them into configurable buckets.
```
Comment (opomuc) [inline README.md:89]: `Histogram arranges observed data by an array of configurable buckets.` -- something along the lines maybe?

File: README.md [hunk 6]
```diff
@@ -81,7 +105,15 @@ local latency = math.random(1, 10)
 http_requests_latency_hist:observe(latency)
 ```
 
-Using summaries:
+### Summary
+
+Summary samples observations and aggregates them into configurable quantiles.
```
Comment (opomuc) [inline README.md:110]: same here, maybe: `Summary aggregates observed data into configurable quantiles.`?

File: README.md [hunk 7]
```diff
@@ -81,7 +105,15 @@ local latency = math.random(1, 10)
...
+Summaries are useful as a service level indicator (e.g. SLAs, SLOs).
+Summary exposed as multiple numerical values:
+- the total count of observed events
+- the total sum of observed values
+- per quantile number of observed events
```
Comment (opomuc) [inline README.md:115]: better to swap: `number of observed event per quantile`

File: README.md [hunk 8]
```diff
@@ -68,7 +84,15 @@ metrics.register_callback(function()
...
 
-Using histograms:
+### Histogram
+
+Histogram samples observations and counts them into configurable buckets.
+Histograms are useful for tracking request latencies, processing time.
+Histogram is exposed as multiple numerical values: 
+- the total count of observed events
+- the total sum of observed values
+- per bucket counters of observed events
```
Comment (opomuc) [inline README.md:94]: `counters of observed events per bucket`

File: README.md (hunk 5 above)
Comment (vasiliy-t) [reply README.md:89]: Nope, it only stores counts, not actual data

File: README.md (hunk 1 above)
Comment (vasiliy-t) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (vasiliy-t) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (vasiliy-t) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (vasiliy-t) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (vasiliy-t) [reply README.md]: done

File: README.md (hunk 7 above)
Comment (vasiliy-t) [reply README.md]: done

File: README.md (hunk 8 above)
Comment (vasiliy-t) [reply README.md]: done

File: README.md [hunk 9]
```diff
@@ -81,7 +105,15 @@ local latency = math.random(1, 10)
 http_requests_latency_hist:observe(latency)
 ```
 
-Using summaries:
+### Summary
+
+Summary aggregates observed values into configurable quantiles.
+Summaries are useful as a service level indicator (e.g. SLAs, SLOs).
+Summary exposed as multiple numerical values:
```
Comment (lenkis) [inline README.md:112]:
```suggestion
Summary is exposed as multiple numerical values:
```

File: README.md [hunk 10]
```diff
@@ -81,7 +105,15 @@ local latency = math.random(1, 10)
...
+- the total count of observed events
+- the total sum of observed values
+- number of observed event per quantile
```
Comment (lenkis) [inline README.md:115]:
```suggestion
- number of observed events per quantile
```

File: README.md [hunk 11]
```diff
@@ -34,15 +43,17 @@ export plugins:
...
+The exact semantics of each metric follows the [prometheus metric types](https://prometheus.io/docs/concepts/metric_types/).
 
-Below are examples of using metrics primitives.
+### Counter
 
-Note that this usage is independent of export-plugins such as
-Prometheus / Graphite / etc.
-For documentation on plugins usage, see [this](https://www.tarantool.io/en/doc/latest/book/monitoring/plugins).
+Counter is a cummulative metric which value can only be incremented or reset to zero on restart.
+Counters are useful for accumulating number of events e.g. requests processed, orders in eshop.
```
Comment (lenkis) [inline README.md:54]:
```suggestion
Counters are useful for accumulating number of events, e.g. requests processed, orders in e-shop. 
```

File: README.md [hunk 12]
```diff
@@ -1,15 +1,24 @@
...
+- exporters to expose collected metrics in Prometheus, Graphite and generic JSON format
+- module to integrate into Tarantool Cartridge based applications
 
-Contents:
+## Table of contents
 
 * [Installation](#installation)
-* [Plugins export](#plugins-export)
-* [Examples](#examples)
+* [Exporters](#plugins-export)
```
Comment (lenkis) [inline README.md:16]: This section's title is different in TOC and in text (Exporters vs Plugins export). It's confusing. Would be nice to choose a single title (whichever you like).

**Review by lenkis — CHANGES_REQUESTED** [review]

File: README.md (hunk 12 above)
Comment (vasiliy-t) [reply README.md]: good point, done

**Review by lenkis — APPROVED** [review]

**Pull request merged**